Sources/Basic/OutputByteStream.swift

```diff
@@ -48,62 +50,94 @@ public class OutputByteStream: OutputStream {
         self.buffer = []
     }
 
+    var bufferSize: Int {
+        return 1024
+    }
+
     // MARK: Data Access API
 
     /// The current offset within the output stream.
     public var position: Int {
         return buffer.count
     }
 
-    /// The contents of the output stream.
-    ///
-    /// This method implicitly flushes the stream.
-    public var bytes: ByteString {
-        flush()
-        return ByteString(self.buffer)
-    }
-    
     // MARK: Data Output API
 
     public func flush() {
-        // Do nothing.
+        writeImpl(buffer)
+        buffer.removeAll(keepingCapacity: true)
+    }
+
+    /// For subclasses to implement. This will be called whenever stream is ready to write
+    /// to the underlying device and buffered will be cleared at that point.
+    func writeImpl(_ bytes: [UInt8]) {
+        fatalError("This should be implemented by subclasses.")
     }
 
     /// Write an individual byte to the buffer.
     public func write(_ byte: UInt8) {
-        buffer.append(byte)
+        write([byte])
     }
     
     /// Write a sequence of bytes to the buffer.
     public func write(_ bytes: [UInt8]) {
+        // This is based on LLVM's raw_ostream.
+        let availableBufferSize = bufferSize - buffer.count
```

Comment:
This should probably be based off of the buffer capacity, not the `bufferSize` property (which should probably just be a static constant for the default buffer capacity).